test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/ProxyEndToEndTests.cs

```diff
@@ -0,0 +1,451 @@
+
+
+        public class TestFixture : IDisposable
+        {
+            private readonly TestServer _testServer;
+            private readonly string _testHome;
+
+            public TestFixture()
+            {
+                // copy test files to temp directory, since accessing the metadata APIs will result
+                // in file creations (for test data files)
+                var scriptSource = Path.Combine(Environment.CurrentDirectory, @"..\..\..\TestScripts\Proxies");
+                _testHome = Path.Combine(Path.GetTempPath(), @"ProxyTests");
+                var scriptRoot = Path.Combine(_testHome, @"site\wwwroot");
+                FileUtility.CopyDirectory(scriptSource, scriptRoot);
+
+                HostOptions = new ScriptApplicationHostOptions
+                {
+                    IsSelfHost = true,
+                    ScriptPath = scriptRoot,
+                    LogPath = Path.Combine(_testHome, @"LogFiles\Application\Functions"),
+                    SecretsPath = Path.Combine(_testHome, @"data\Functions\Secrets"),
+                    TestDataPath = Path.Combine(_testHome, @"data\Functions\SampleData")
+                };
+
+                FileUtility.EnsureDirectoryExists(HostOptions.TestDataPath);
+
+                var optionsMonitor = TestHelpers.CreateOptionsMonitor(HostOptions);
+
+                var workerOptions = new LanguageWorkerOptions
+                {
+                    WorkerConfigs = TestHelpers.GetTestWorkerConfigs()
+                };
+
+                var provider = new HostFunctionMetadataProvider(optionsMonitor, NullLogger<HostFunctionMetadataProvider>.Instance, new TestMetricsLogger());
+
+                var builder = AspNetCore.WebHost.CreateDefaultBuilder()
+                   .UseStartup<Startup>()
+                   .ConfigureServices(services =>
+                   {
+                       services.Replace(new ServiceDescriptor(typeof(IOptions<ScriptApplicationHostOptions>), new OptionsWrapper<ScriptApplicationHostOptions>(HostOptions)));
+                       services.Replace(new ServiceDescriptor(typeof(ISecretManagerProvider), new TestSecretManagerProvider(new TestSecretManager())));
+                       services.Replace(new ServiceDescriptor(typeof(IOptionsMonitor<ScriptApplicationHostOptions>), optionsMonitor));
+                       services.Replace(new ServiceDescriptor(typeof(IFunctionMetadataProvider), provider));
+                   });
+
+                // TODO: https://github.com/Azure/azure-functions-host/issues/4876
+                _testServer = new TestServer(builder);
+                HostOptions.RootServiceProvider = _testServer.Host.Services;
+                var scriptConfig = _testServer.Host.Services.GetService<IOptions<ScriptJobHostOptions>>().Value;
+
+                HttpClient = _testServer.CreateClient();
+                HttpClient.BaseAddress = new Uri("https://localhost/");
+
+                TestHelpers.WaitForWebHost(HttpClient);
```

Comment:
Syncing your branch, I see that all the tests in this suite are failing because the TestFixture isn't set up properly (the tests have been disabled in this branch for some time, so the code is stale). Replace these highlighted lines with the following:

```csharpTestHost = new TestFunctionHost(HostOptions.ScriptPath, HostOptions.LogPath,
    configureScriptHostServices: services =>
    {
        services.Replace(new ServiceDescriptor(typeof(IOptions<ScriptApplicationHostOptions>), new OptionsWrapper<ScriptApplicationHostOptions>(HostOptions)));
        services.Replace(new ServiceDescriptor(typeof(ISecretManagerProvider), new TestSecretManagerProvider(new TestSecretManager())));
        services.Replace(new ServiceDescriptor(typeof(IOptionsMonitor<ScriptApplicationHostOptions>), optionsMonitor));
        services.Replace(new ServiceDescriptor(typeof(IFunctionMetadataProvider), provider));
    });

HttpClient = TestHost.HttpClient;

TestHelpers.WaitForWebHost(HttpClient);
```

Remove the unused _testServer field, and change Dispose to dispose